sig-network/plumbing-working-group/[v1] Kubernetes Network Custom Resource Definition De-facto Standard.md

```diff
@@ -0,0 +1,631 @@
+
+The interaction of additional pod network attachments with the Kubernetes API and its objects, such as Services, Endpoints, proxies, etc is not specified. SIG Network discussed this topic at length in July/August 2017 and decided it would require changes to the Kubernetes API, which is an explicit non-goal of this specification.
+
+1.1.4 Changes to the Kubernetes CNI Driver
+
+To ensure easy use of this specification and implementations of it, no changes will be required of the Kubernetes CNI driver (eg pkg/kubelet/dockershim/network/cni/*).  Changes to the driver to support multiple pod network attachments have already been proposed and multiple proof-of-concepts written, but there is as yet no upstream consensus on how multiple attachments should work and thus what changes would be required in the CNI driver.
+
+1.1.5 Replacement of the Kubernetes Cluster-Wide Default Network
+
+To preserve compatibility and expectations with existing Kubernetes deployments, this specification does not attempt to change any of the existing Kubernetes cluster-wide network behavior.  All pods must still be attached to the cluster-wide default network as they are today.
+
+1.1.6 Enabling Implementations That Are Not CNI Plugins
+
+This specification only attempts to enable Kubernetes network plugins which use the CNI driver of a Kubernetes runtime like dockershim, CRI-O, and rkt.  This specification expects pod network attachment operations to be directed through the implementation, which is required to be a CNI plugin.  Future versions of this specification may revisit this requirement.
+
+**2. Definitions**
+
+2.1 Implementation
+
+The Kubernetes network plugin that implements this specification; as of Kubernetes 1.11 this is defined to be a plugin conforming to the [Container Network Interface](https://github.com/containernetworking/cni/blob/master/SPEC.md) (CNI) specification v0.1.0 or later.  Kubernetes should be configured to call the implementation for all pod network operations, and the implementation then determines which additional operations to perform based on the pods annotations and the Custom Resources defined in this specification.
+
+2.2 Kubernetes Cluster-Wide Default Network
+
+A network to which all pods are attached following the current behavior and requirements of Kubernetes.
+
+2.3 Network Attachment
+
+A means of allowing a pod to directly communicate with a given logical or physical network. Typically (but not necessarily) each attachment takes the form of a kernel network interface placed in to the pods network namespace.  Each attachment may result in zero or more IP addresses being assigned to the pod.
+
+2.4 CNI Delegating Plugin
+
+An implementation conforming to this specification which delegates pod network attachment/detachment operations to other plugins conforming to the CNI specification.  Examples include [Multus](https://github.com/intel/multus-cni) and [CNI-Genie](https://github.com/Huawei-PaaS/CNI-Genie).
```

Comment:
Thanks @dcbw 

CNI-Genie team will be happy to implement the multi-network Specification.